crates/algokit_transact/src/transactions/asset_transfer.rs

```diff
@@ -1,37 +1,70 @@
+//! Asset transfer transaction module for Algokit Core.
+//!
+//! This module provides functionality for creating and managing asset transfer transactions.
+
 use crate::address::Address;
 use crate::traits::{AlgorandMsgpack, TransactionId};
 use crate::transactions::common::TransactionHeader;
 use crate::utils::{is_zero, is_zero_addr, is_zero_addr_opt};
 use serde::{Deserialize, Serialize};
 use serde_with::{serde_as, skip_serializing_none};
 
+/// Represents an asset transfer transaction that moves ASAs between accounts.
+///
+/// Asset transfer transactions are used to transfer Algorand Standard Assets (ASAs)
+/// from one account to another.
 #[serde_as]
 #[skip_serializing_none]
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub struct AssetTransferTransactionFields {
+    /// Common transaction header fields.
     #[serde(flatten)]
     pub header: TransactionHeader,
 
+    /// The ID of the asset being transferred.
     #[serde(rename = "xaid")]
     #[serde(skip_serializing_if = "is_zero")]
     #[serde(default)]
     pub asset_id: u64,
 
+    /// The amount of the asset to transfer.
+    ///
+    /// An integer value representing the number of units (to their smallest denomination) of
+    /// the asset that are being transferred.
+    /// In other words, the asset decimals don't play a role in this value.
+    /// It should be up to the caller (or a higher abstraction) to handle the conversion based on
+    /// the asset decimals.
     #[serde(rename = "aamt")]
     #[serde(skip_serializing_if = "is_zero")]
     #[serde(default)]
     pub amount: u64,
 
+    /// The address of the account that will receive the asset.
+    ///
+    /// The receiver must have opted-in to the asset before they can receive it.
     #[serde(rename = "arcv")]
     #[serde(skip_serializing_if = "is_zero_addr")]
     #[serde(default)]
     pub receiver: Address,
 
+    /// Optional address of the account that actually holds the asset being transferred.
+    ///
+    /// If provided, this indicates that the transaction is a clawback operation,
+    /// where the sender is the asset clawback address and is forcibly moving assets
+    /// from this account to the receiver.
+    // FIXME: This description is untouched from what Claude wrote.
+    // We should assess if asset_sender should be part of a basic asset transfer (probably not).
```

Comment:
I feel like we want to represent all the various raw transaction fields at the core level (like it currently does) and let higher level abstractions like utils take care of splitting based on use case.

Reply:
I would definitely want both be represented. Not sure if utils is the best place for this higher abstraction because I would want it to be where most users are consuming the crates and not make them so optional.
What I mean is that I see this higher abstraction being the default, while the actual representation being something you reach for only in rare cases.
Granted, both should exist.

Reply:
Thoughts @lempira and @joe-p.